lib/src/parsers/type-parser.ts

```diff
@@ -455,6 +459,30 @@ function parseUnionType(
   }
 }
 
+/**
+ * Parse an intersection type node.
+ *
+ * @param typeNode union type node
+ * @param typeTable a TypeTable
+ * @param lociTable a LociTable
+ */
+function parseIntersectionTypeNode(
```

Comment:
I'm thinking we should limit intersection type members to objects only. I can't see any use-case for intersection types that involve non-objects (e.g. `string & "a"`). This will make working with intersection types a bit easier.